vm/src/vm/vm_memory/memory.rs

```diff
@@ -676,6 +757,7 @@ impl<'a> RelocateValue<'a, &'a MaybeRelocatable, Cow<'a, MaybeRelocatable>> for
         Ok(match value {
             MaybeRelocatable::Int(_) => Cow::Borrowed(value),
             MaybeRelocatable::RelocatableValue(addr) => {
+                #[allow(clippy::useless_conversion)]
                 Cow::Owned(self.relocate_value(*addr)?.into())
```

Comment:
```suggestion
                Cow::Owned(self.relocate_value(*addr)?)
```

Reply:
This results in the following error:

```
error[E0271]: type mismatch resolving `<MaybeRelocatable as ToOwned>::Owned == Relocatable`
   --> vm/src/vm/vm_memory/memory.rs:760:17
    |
760 |                 Cow::Owned(self.relocate_value(*addr)?)
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `MaybeRelocatable`, found `Relocatable`
```

because `fn relocate_value()` now has a different fn signature (return type is either `Relocatable` or `MaybeRelocatable` depending on whether feature `"extensive_hints"` is on or not (respectively)).

Reply:
Ok, you're right. So you can do this instead:

```suggestion
                #[cfg(not(feature = "extensive_hints"))]
                let v = self.relocate_value(*addr)?.into();
                #[cfg(feature = "extensive_hints")]
                let v = self.relocate_value(*addr)?;

                Cow::Owned(v)
```

I think this way shows the reason why you need to call `into()`.

Reply:
Ok, good point, this makes it more clear what's going on :+1:

Added in 72d27c8